packages/edge-login-ui-rn/src/components/screens/newAccount/NewAccountPinScreen.js

```diff
@@ -1,180 +1,158 @@
+  username,
+  password,
+  pin,
+  createUser,
+  pinErrorMessage,
+  createErrorMessage,
+  clearCreateErrorMessagecircleFilled,
+  theme
+}: Props) => {
+  const [isProcessing, setIsProcessing] = useState(false)
+  const styles = getStyles(theme)
 
-  render() {
-    this.checkError()
-    return (
-      <SafeAreaView>
-        <View style={styles.screen}>
-          <Header onBack={this.props.onBack} />
-          <View style={styles.pageContainer}>
-            <View style={styles.row1}>
-              <Text style={styles.instructions}>{s.strings.pin_desc}</Text>
-            </View>
-            <View style={styles.row2}>
-              <FourDigitInput testID="pinInput" />
-            </View>
-            <View style={styles.row3}>
-              <Button
-                testID="nextButton"
-                onPress={this.handleNext}
-                downStyle={styles.nextButton.downStyle}
-                downTextStyle={styles.nextButton.downTextStyle}
-                upStyle={styles.nextButton.upStyle}
-                upTextStyle={styles.nextButton.upTextStyle}
-                label={s.strings.next_label}
-                isThinking={this.state.isProcessing}
-                doesThink
-              />
-            </View>
-          </View>
-        </View>
-      </SafeAreaView>
+  if (createErrorMessage) {
+    Alert.alert(
+      s.strings.create_account_error_title,
+      s.strings.create_account_error_message + '\n' + createErrorMessage,
+      [{ text: s.strings.ok }]
     )
+    clearCreateErrorMessagecircleFilled()
   }
 
-  handleNext = () => {
-    this.setState({
-      isProcessing: true,
-      createErrorMessage: null
-    })
+  const handleNext = () => {
+    setIsProcessing(true)
+
     // validation.
-    // is there no error message ,
-    if (this.props.pin.length !== 4 || this.props.pinError) {
-      this.setState({
-        isProcessing: false
-      })
+    // is there no error message,
+    if (pin.length !== 4 || pinErrorMessage) {
+      setIsProcessing(false)
       logEvent(`Signup_PIN_Invalid`)
       return
     }
+
     logEvent(`Signup_Create_User`)
-    this.props.createUser({
-      username: this.props.username,
-      password: this.props.password,
-      pin: this.props.pin
+    createUser({
+      username,
+      password,
+      pin
     })
   }
+
+  return (
+    <ThemedScene>
+      <SimpleSceneHeader>{s.strings.create_your_account}</SimpleSceneHeader>
+      <View style={styles.content}>
+        <EdgeText
+          style={styles.subtitle}
+        >{`${s.strings.step_three}: ${s.strings.set_four_digit_pin}`}</EdgeText>
+        <EdgeText style={styles.description} numberOfLines={2}>
+          {s.strings.pin_desc}
+        </EdgeText>
+        <FourDigitInput />
+        <FormError>{pinErrorMessage}</FormError>
+        <View style={styles.actions}>
+          <Fade
+            visible={
+              pin.length === 4 && !pinErrorMessage && !createErrorMessage
```

Comment:
Avoid using numbers directly in the code.
Always pass them as a Param/Constant.